load-test-framework/src/main/java/com/google/pubsub/flic/controllers/GCEController.java

```diff
@@ -67,24 +67,41 @@
    */
   private GCEController(String projectName, Map<String, Map<ClientParams, Integer>> types,
                         ScheduledExecutorService executor, Storage storage,
-                        Compute compute, PubSub pubSub, int cpsPublisherCount,
-                        int cpsSubscriberCount, int kafkaPublisherCount,
-                        int kafkaSubscriberCount) throws Throwable {
+                        Compute compute, PubSub pubSub) throws Throwable {
     super(executor);
     this.projectName = projectName;
     this.types = types;
     this.storage = storage;
     this.compute = compute;
-    this.cpsPublisherCount = cpsPublisherCount;
-    this.cpsSubscriberCount = cpsSubscriberCount;
-    this.kafkaPublisherCount = kafkaPublisherCount;
-    this.kafkaSubscriberCount = kafkaSubscriberCount;
+    cpsPublisherCount = 0;
+    cpsSubscriberCount = 0;
+    kafkaPublisherCount = 0;
+    kafkaSubscriberCount = 0;
 
-    // For each unique type of CPS Publisher, create a Topic if it does not already exist, and then
-    // delete and recreate any subscriptions attached to it so that we do not have backlog from
-    // previous runs.
     List<SettableFuture<Void>> pubsubFutures = new ArrayList<>();
-    types.values().forEach((paramsMap) -> paramsMap.keySet().stream().map(p -> p.clientType)
+    types.values().forEach((paramsMap) -> {
+      // Iterate through map to aggregate subscriber and publisher counts
+      paramsMap.forEach((param, count) -> {
+        switch (param.clientType) {
+          case CPS_GCLOUD_PUBLISHER:
+            cpsPublisherCount += count;
+            break;
+          case CPS_GCLOUD_SUBSCRIBER:
+            cpsSubscriberCount += count;
+            break;
+          case KAFKA_PUBLISHER:
+            kafkaPublisherCount += count;
+            break;
+          case KAFKA_SUBSCRIBER:
+            kafkaSubscriberCount += count;
+            break;
+        }
+      });
```

Comment:
How about:
`cpsPublisherCount = types.values().forEach(paramsMap -> paramsMap.keySet().stream().map(p -> p.clientType).filter(t -> t == CPS_GCLOUD_PUBLISHER).count()` instead of this loop